subsys/bluetooth/audio/broadcast_sink.c

```diff
@@ -0,0 +1,1033 @@
+	if (bis->index == 0) {
+		BT_DBG("BIS index was 0");
+		return false;
+	}
+
+	/* codec config data length */
+	len = net_buf_simple_pull_u8(buf);
+	if (len > buf->len) {
+		BT_DBG("Invalid BIS specific codec config data length: "
+		       "%u (buf is %u)", len, buf->len);
+		return false;
+	}
+
+	if (len > 0) {
+		struct net_buf_simple ltv_buf;
+		void *ltv_data;
+
+		if (codec_data_already_found) {
+			/* Codec config can either be specific to each
+			 *  BIS or for all, but not both
+			 */
+			BT_DBG("BASE contains both codec config data and BIS "
+			       "codec config data. Aborting.");
+			return false;
+		}
+
+		/* TODO: Support codec configuration data per bis */
+		BT_WARN("BIS specific codec config data of length %u "
+			"was found but is not supported yet", len);
+
+		/* Use an extra net_buf_simple to be able to decode until it
+		 * is empty (len = 0)
+		 */
+		ltv_data = net_buf_simple_pull_mem(buf, len);
+		net_buf_simple_init_with_data(&ltv_buf, ltv_data, len);
+
+		while (ltv_buf.len != 0) {
+			struct bt_codec_data *bis_codec_data;
+
+			bis_codec_data = &bis->data[bis->data_count];
+
+			if (!net_buf_decode_codec_ltv(&ltv_buf,
+						      bis_codec_data)) {
+				BT_DBG("Failed to decode BIS config data for entry %u",
+				       bis->data_count);
+				return false;
+			}
+			bis->data_count++;
+		}
+	}
+
+	return true;
+}
+
+static bool net_buf_decode_subgroup(struct net_buf_simple *buf,
+				    struct bt_audio_base_subgroup *subgroup)
+{
+	struct net_buf_simple ltv_buf;
+	struct bt_codec	*codec;
+	void *ltv_data;
+	uint8_t len;
+
+	codec = &subgroup->codec;
+
+	subgroup->bis_count = net_buf_simple_pull_u8(buf);
+	if (subgroup->bis_count > ARRAY_SIZE(subgroup->bis_data)) {
+		BT_DBG("BASE has more BIS %u than we support %u",
+		       subgroup->bis_count,
+		       (uint8_t)ARRAY_SIZE(subgroup->bis_data));
+		return false;
+	}
+	codec->id = net_buf_simple_pull_u8(buf);
+	codec->cid = net_buf_simple_pull_le16(buf);
+	codec->vid = net_buf_simple_pull_le16(buf);
+
+	/* codec configuration data length */
+	len = net_buf_simple_pull_u8(buf);
+	if (len > buf->len) {
+		BT_DBG("Invalid codec config data length: %u (buf is %u)",
+		len, buf->len);
+		return false;
+	}
+
+	/* Use an extra net_buf_simple to be able to decode until it
+	 * is empty (len = 0)
+	 */
+	ltv_data = net_buf_simple_pull_mem(buf, len);
+	net_buf_simple_init_with_data(&ltv_buf, ltv_data, len);
+
+	/* The loop below is very similar to codec_config_store with notable
+	 * exceptions that it can do early termination, and also does not log
+	 * every LTV entry, which would simply be too much for handling
+	 * broadcasted BASEs
+	 */
+	while (ltv_buf.len != 0) {
+		struct bt_codec_data *codec_data = &codec->data[codec->data_count++];
+
+		if (!net_buf_decode_codec_ltv(&ltv_buf, codec_data)) {
+			BT_DBG("Failed to decode codec config data for entry %u",
+			       codec->data_count - 1);
+			return false;
+		}
+	}
+
+	if (sizeof(len) > buf->len) {
```

Comment:
The convention in most of the tree (and which I also find more intuitive to read) is to have the variable on the left side and the constant on the right side of compatisons.